Money instance creation from a formatted string (#16)
archtechx/money: README.md

$money->formatted(['decimalSeparator' => ',', 'prefix' => '$ ', 'suffix' => ' USD']);
```

> There is also a `->rawFormatted()` if you wish to use [math decimals](#math-decimals) instead of [display decimals](#display-decimals).

Reviewer: ```suggestion
There's also `->rawFormatted()` if you wish to use [math decimals](#math-decimals) instead of [display decimals](#display-decimals).
```

```php

> $money = Money::new(123456, CZK::class);
> $money->rawFormatted(); // 1 235,56 Kč

Reviewer: ```suggestion
$money = Money::new(123456, CZK::class);
$money->rawFormatted(); // 1 234,56 Kč
```

```

> And converting the formatted value back to the Money instance is also possible. The package tries to extract the currency from the provided string.

Reviewer: ```suggestion
Converting the formatted value back to the `Money` instance is also possible. The package tries to extract the currency from the provided string:
```

```php
$money = money(1000);
$formatted = $money->formatted(); // $10.00
$fromFormatted = Money::fromFormatted($formatted);
$fromFormatted->is($money); // true
```

Optional overrides for the [currency specification](#currency-logic) are accepted too.
```php
$money = money(1000);
$formatted = $money->formatted(); // $10.00

> $fromFormatted = Money::fromFormatted($formatted, USD::class, ['decimalSeparator' => ',', 'prefix' => '$ ', 'suffix' => ' USD']);

Reviewer: What is the array here for? When the formatted string is `$10.00`

Author: Optionally the user can override the config. This section is similar to the [Formatting money](https://github.com/archtechx/money#formatting-money) section.

![image](https://user-images.githubusercontent.com/11887260/158368520-1e5fe294-921a-4a52-98dd-840b57ff539d.png)

Reviewer: > Here is an example of when the user may need this

Right, so it's essentially for specifying the *source* currency format that's used by the parsed string.

In the example above, that's essentially a "union" of `USD::class` and the `['prefix' => ...]` array, right?

So the original string would be in that format, and the newly created `Money` instance would *also* be in that mixed format with overwritten `USD`? Or would the new instance just use `USD`?

Author: > the newly created Money instance would also be in that mixed format with overwritten USD? Or would the new instance just use USD?

The new instance would just use USD. The overrides do not apply to the new money instance.

And I believe that is the expected behavior. If the user wants to have the instance in a `custom` currency, he/she should have created a custom currency from the beginning before formatting and unformatting it.

Reviewer: Right, I see. Makes sense. So the last array just provides context for what was used in `->formatted()`.
